Sources/xcodeproj/Project/XcodeProj.swift

```diff
@@ -69,6 +69,22 @@ public final class XcodeProj: Equatable {
             lhs.pbxproj == rhs.pbxproj &&
             lhs.sharedData == rhs.sharedData
     }
+    
+    // MARK: - Private
+    
+    private static func readPBXProj(url: URL) throws -> (Data, [String: Any]) {
+        let plistXML = try Data(contentsOf: url)
+        var propertyListForamt =  PropertyListSerialization.PropertyListFormat.xml
+        let serialized = try PropertyListSerialization.propertyList(
+            from: plistXML,
+            options: .mutableContainersAndLeaves,
+            format: &propertyListForamt
+        )
+        guard let dictionary = serialized as? [String: Any] else {
+            return (plistXML, [:])
```

Comment:
Am I wrong in thinking that if the serialised object is not a `[String: Any]` the performance improvements that you introduced will not be executed? I think it's safe to use `as!` here. I doubt that Xcode changes its format to have a key other than a string.